### DISCUSSION TITLE
How to get back after gd (goto definition)?

### DISCUSSION THREAD

**Discussion by DrakorissVere** [VonHeikemen/lsp-zero.nvim#373] [Q&A] [answered]
Hi all!

`lsp-zero` is great, I'm enjoying it so far. Thanks for developing it!
As for a question, I have a particular case that I can't resolve on my own, and it's kind of annoying to handle coding w/o it: after inspecting some definitions with `gd` I want to return to where I called `gd`, but I can't find how. Is it even possible?

Also, a small question that may not even related to `lsp-zero`:
Let's say I accepted the `C-y` autocompletion variant for a method with 2 args:
```rust
.build(name, level)
```
Method expanded, cursor appeared on 'name' and I can freely start typing whatever I want. But after that, I want to jump onto the next arg and replace it. I know I can do it in `neovim` (I'm still a newbie), like exit insert mode, jump word forward, delete word, and enter insert mode again... But maybe there's some other way around it? Does `lsp-zero` autocompletion provide such utility or do I have to manage it through neovim entirely?

Thanks in advance!

**Comment by VonHeikemen** [comment, marked as answer]
`Ctrl + t` should take you back to the position where you called `gd`.

You can add a custom keybinding to [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) (the autocompletion plugin) to navigate between snippet placeholders.

Here is an example.

 ```lua
local cmp = require('cmp')
local cmp_action = require('lsp-zero').cmp_action()

cmp.setup({
  mapping = cmp.mapping.preset.insert({
    ['<C-f>'] = cmp_action.luasnip_jump_forward(),
    ['<C-b>'] = cmp_action.luasnip_jump_backward(),
  })
})
```

**Comment by DrakorissVere** [comment]
It works! Thank you.
